ansible/roles/operator-pipeline/templates/openshift/tasks/get-ci-reviewer.yml

```diff
@@ -0,0 +1,79 @@
+    - name: github_token_secret_key
+      description: The key within the Kubernetes Secret that contains the GitHub token.
+      default: token
+    - name: github_token_secret_name
+      description: The name of the Kubernetes Secret that contains the GitHub token.
+      default: github
+  workspaces:
+    - name: repo
+      description: Cloned repository
+  results:
+    - name: author_is_reviewer
+      description: >-
+        Check if the author of the PR is listed also as a reviewer
+    - name: list_reviewers
+      description: >-
+        Return list of all reviewers from ci.yaml file except author of the pull request
+  steps:
+    - name: get-ci-reviewer
+      image: "$(params.pipeline_image)"
+      workingDir: $(workspaces.repo.path)
+      script: |
+        #! /usr/bin/env bash
+        set -e
+        CI="operators/$(params.operator_name)/ci.yaml"
+        if [ ! -e "$CI" ]; then
+          echo >$(results.list_reviewers.path)
+          echo false >$(results.author_is_reviewer.path)
+        else
+          if grep -q 'reviewers:' $CI; then
+            yq -r '.reviewers[]' $CI >$(results.list_reviewers.path)
+            yq -r ".reviewers|index(\"$(params.git_username)\") != null" $CI >$(results.author_is_reviewer.path)
+          else
+            echo >$(results.list_reviewers.path)
+            echo false >$(results.author_is_reviewer.path)
+          fi
+        fi
+    - name: approve-pull-request
+      env:
+        - name: GITHUB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.github_token_secret_name)
+              key: $(params.github_token_secret_key)
+      image: "$(params.pipeline_image)"
+      script: |
+        #! /usr/bin/env bash
+        set -e
+        # do not use -x to not leak github secrets into logs!
+
+        AUTHOR_IS_REVIEWER=$(cat $(results.author_is_reviewer.path))
+        CI_REVIEWERS=$(cat $(results.list_reviewers.path))
+        # requesting review from repo maintainers if there are no reviewers in ci.yaml or no ci.yaml file
+        if [ -z "$CI_REVIEWERS" ]; then
+          gh pr edit $(params.pr_url) --add-reviewer redhat-openshift-ecosystem/pipeline-maintainers
+          echo "Reviewers are missing in ci.yaml file - requesting review from repo maintainers."
+        else
+          # requesting review from PR author if it is included in the reviewers field
+          if [ "$AUTHOR_IS_REVIEWER" = true ]; then
+            gh pr review $(params.pr_url) --approve
+            echo "This pull request was automatically approved due to PR author is in the reviewers list."
+          else
+            # request a review from the reviewers listed in ci.yml file if the author is not listed in ci.yaml
+            for reviewer in $CI_REVIEWERS; do
+              gh pr edit $(params.pr_url) --add-reviewer `echo $reviewer | xargs | tr ' ' ','`
+              echo "Requesting review from reviewers in ci.yaml file: $reviewer"
+            done
```

Comment:
The purpose of the `xargs | tr ' ' ','` part in my comment was to avoid the for loop and add all the reviewers in a sigle call to `gh`.
It's not very clear from the `gh` documentation but if you look at the examples, `--add-reviewer` accepts multiple usernames separated by commas, so you should be able to just do:

```shell
gh pr edit $(params.pr_url) --add-reviewer $(echo $CI_REVIEWERS | xargs | tr ' ' ','`)
```

or, if you prefer:

```
gh pr edit $(params.pr_url) --add-reviewer $(echo $CI_REVIEWERS | paste -d, -s)
```